### PR TITLE
fix: package json exports

### LINKED ISSUE #71: improper package.json exports
### Environment

node v22.2.0
pnpm v9.1.3

### Reproduction

https://publint.dev/serve-placeholder@2.0.1

https://arethetypeswrong.github.io/?p=serve-placeholder%402.0.1

### Describe the bug

I installed the package and imported it, and got a typescript error. The first thing I checked was these 2 places to verify types export validity:

- https://publint.dev/serve-placeholder@2.0.1

- https://arethetypeswrong.github.io/?p=serve-placeholder%402.0.1

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by o-az**
### 🔗 Linked issue

#71

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Resolves #71

- Update `package.json#exports` to provide complete support for CJS & ESM,
 
### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,16 @@
...
-      "require": "./dist/index.cjs"
+      "import": {
+        "types": "./dist/index.d.mts",
+        "import": "./dist/index.mjs",
+        "default": "./dist/index.mjs"
+      },
+      "require": {
+        "types": "./dist/index.d.cts",
+        "require": "./dist/index.cjs",
+        "default": "./dist/index.cjs"
```
Comment (pi0) [inline package.json:19]: Can you please limit changes in this PR only to this? 🙏🏼

Comment (o-az) [reply package.json:19]: done

File: package.json [hunk 2]
```diff
@@ -8,8 +8,16 @@
   "type": "module",
   "exports": {
     ".": {
-      "import": "./dist/index.mjs",
-      "require": "./dist/index.cjs"
+      "import": {
+        "types": "./dist/index.d.mts",
```
Comment (danielroe) [inline package.json:12]: I think once `unbuild` is updated in this repo then this explicit `types` line will not be required.

**Comment by pi0** [comment]
Thanks for PR again ❤️  updated on mian

**Pull request closed without merging**